src/bootloaders/syslinux.c

```diff
@@ -143,7 +146,25 @@ static bool syslinux_set_default_kernel(const BootManager *manager, const Kernel
                 cbm_writer_append_printf(writer, "  KERNEL %s\n", k->target.legacy_path);
                 /* Add the initrd if we found one */
                 if (k->target.initrd_path) {
-                        cbm_writer_append_printf(writer, "  INITRD %s\n", k->target.initrd_path);
+                        cbm_writer_append_printf(writer, "  INITRD %s", k->target.initrd_path);
+                        boot_manager_initrd_iterator_init(manager, &iter);
+                        while (boot_manager_get_initrd_freestanding_entry(&iter, initrd_name)) {
+                                cbm_writer_append_printf(writer, ",%s", initrd_name);
+                        }
+                        cbm_writer_append_printf(writer, "\n");
+                } else {
+                        boot_manager_initrd_iterator_init(manager, &iter);
+                        while (boot_manager_get_initrd_freestanding_entry(&iter, initrd_name)) {
+                                if (initrd_counter == 0) {
+                                        cbm_writer_append_printf(writer, "  INITRD %s", initrd_name);
+                                } else {
+                                        cbm_writer_append_printf(writer, ",%s", initrd_name);
+                                }
+                                initrd_counter++;
+                        }
+                        if (initrd_counter) {
+                                cbm_writer_append_printf(writer, "\n");
+                        }
```

Comment:
Same as in comment to grub config writing. Make writing the global initrds a separate step as opposed to duplicated code in `if/else` statement.